app/src/main/kotlin/io/homeassistant/companion/android/widgets/BaseGlanceEntityWidgetReceiver.kt

```diff
@@ -0,0 +1,289 @@
+        deleteWidgetsFromDatabase(appWidgetIds)
+    }
+
+    /**
+     * Register this receiver to receive [Intent.ACTION_SCREEN_ON] and [Intent.ACTION_SCREEN_OFF].
+     * It doesn't exported the receiver.
+     */
+    fun registerReceiver(context: Context) {
+        Timber.tag(widgetClassName).d("Register receiver ($this)")
+        ContextCompat.registerReceiver(
+            context,
+            this,
+            IntentFilter().apply {
+                addAction(Intent.ACTION_SCREEN_OFF)
+                addAction(Intent.ACTION_SCREEN_ON)
+            },
+            ContextCompat.RECEIVER_NOT_EXPORTED,
+        )
+    }
+
+    private fun deleteWidgetsFromDatabase(appWidgetIds: IntArray) {
+        widgetScope.launch {
+            Timber.tag(widgetClassName).d("onDeleted ${appWidgetIds.toList()}")
+            dao.deleteAll(appWidgetIds)
+            appWidgetIds.forEach(::removeSubscription)
+        }
+    }
+
+    private fun setupWidgetScope() {
+        if (!widgetScope.isActive) {
+            widgetScope = widgetScopeProvider()
+        }
+    }
+
+    private fun watchingForEntitiesChanges(context: Context) {
+        setupWidgetScope()
+        if (!serverManager.isRegistered()) {
+            Timber.tag(widgetClassName).d("No server registered won't watch for entities")
+            return
+        }
+
+        Timber.e("Watching for entities $widgetScope")
+        widgetScope.launch {
+            val entitiesPerServer = cleanupOrphansWidgetAndGetEntitiesByServer(context)
+
+            entitiesPerServer.forEach { (appWidgetId, entitiesPerServer) ->
+                widgetJobs[appWidgetId]?.cancel()
+
+                val serverId = entitiesPerServer.serverId
+                val entitiesId = entitiesPerServer.entityIds
+
+                val entityUpdatesFlow = if (serverManager.getServer(serverId) != null) {
+                    serverManager.integrationRepository(serverId).getEntityUpdates(entitiesId)
+                } else {
+                    null
+                }
+                if (entityUpdatesFlow != null) {
+                    widgetJobs[appWidgetId] = widgetScope.launch {
+                        Timber.tag(widgetClassName).d("Watching updates of entities($entitiesId) for widget $appWidgetId")
+                        entityUpdatesFlow.collect {
+                            onEntityUpdate(context, appWidgetId, it)
+                            updateView(context, appWidgetId)
+                        }
+                    }
+                } else {
+                    Timber.tag(widgetClassName).d("Entity updates is null for widget $appWidgetId not watching")
+                    // Shouldn't do anything since the job should have been canceled already
+                    removeSubscription(appWidgetId)
+                }
+            }
+        }
+    }
+
+    private fun stopWatchingForEntitiesChanges() {
+        Timber.i("Stop watching for ${glanceAppWidget.javaClass}")
```

Comment:
If you're going to be fancy use the fancy variable ;)

```suggestion
        Timber.i("Stop watching for $widgetClassName")
```